### PR TITLE
Some deprecations in advance of Drupal 11

### PR DESCRIPTION
**Pull request by rosiel**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/2318

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?
Removes deprecated code per the following:

* [New API for defining Field Type categories](https://www.drupal.org/node/3375748)
* [EntityReferenceTestTrait has been renamed to EntityReferenceFieldCreationTrait](https://www.drupal.org/node/3401941)
* and updating the cache action version in CI.

# How should this be tested?

* [Run tests locally](https://islandora.github.io/documentation/technical-documentation/running-automated-tests/) (deprecation notices seem to be suppressed in CI). You shouldn't be able to track down these errors to issues in Islandora code (note that not all contrib modules have used the new API for defining field type categories so those may pop up but it seems it's not our fault.)

# Documentation Status

* Does this change existing behaviour that's currently documented? no!
* Does this change require new pages or sections of documentation? no!
* Who does this need to be documented for? nobody!
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers

Draft because there are a lot more problems when running tests locally.

### CODE REVIEW

**Comment by seth-shaw-asu** [comment]
The new Field Type API is 10.2+. The note appears to say we can [add 10.1 support by adding a hook](https://www.drupal.org/node/3375748#bc).

**Comment by rosiel** [comment]
Hmm. With 2 weeks till the release of 10.3, and the end of security support for 10.1.x, I think we should move toward declaring the module now requires 10.2 or greater. (If it makes people feel better, we could wait two weeks and then do this).

**Comment by rosiel** [comment]
add composer conflicts.

File: composer.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
   ],
   "require": {
     "drupal/context": "^4 || ^5@RC",
+    "drupal/core": "^10.2",
```
Comment (jordandukart) [inline composer.json]: Don't think is needed especially with a long term switch to drupal.org https://www.drupal.org/docs/develop/using-composer/add-a-composerjson-file#core-compatibility.

**Comment by jordandukart** [comment]
Rest here I think looks fine.

**Review by adam-vessey — APPROVED** [review]

**Pull request merged**